pkg/client/dummy_clientset.go

```diff
@@ -155,3 +157,25 @@ func (c *DummyOpenPipelineClient) GetAll(ctx context.Context) ([]coreapi.Respons
 func (c *DummyOpenPipelineClient) Update(_ context.Context, _ string, _ []byte) (openpipeline.Response, error) {
 	return openpipeline.Response{}, nil
 }
+
+type DummySegmentClient struct{}
+
+func (c *DummySegmentClient) List(ctx context.Context) (segments.Response, error) {
+	return segments.Response{}, fmt.Errorf("unimplemented")
```

Comment:
I think we need an implementation that just does nothing here, similar to OpenPipeline, as this is also used by dry run.

Currently segment dry run reports: `Deployment failed - Monaco Error: failed to deploy segment with externalId: ..., with error: failed to fetch segment object: unimplemented`

Reply:
This probably applies to all the methods in the dummy client.

Reply:
Maybe we should rename it at some point to dry-run-client

Reply:
@arthurpitman thanks i will split it up between dry-run-client and test client.